clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -267,6 +286,7 @@ protected void onJoinPrepare(int generation, String memberId) {
                     listener.getClass().getName(), groupId, e);
         }
 
+        assignmentSnapshot = null;
```

Comment:
If we reset here, doesn't that mean upon ` onJoinComplete` this variable will always be null for `ConsumerCoordinator`?